Data/HashMap/Internal.hs

```diff
@@ -579,23 +620,31 @@ isLeafOrCollision _               = False
 -- * Construction
 
 -- | /O(1)/ Construct an empty map.
-empty :: HashMap k v
-empty = Empty
+empty :: forall k v . HashMap k v
+empty = empty'
+
+-- | like 'empty' but allows a custom salt to be set
+empty' :: forall k v (salt :: Nat) . HashMapT salt k v
```

Comment:
I believe the salt should be the first type argument for use with `TypeApplications`.

Reply:
Good point, for new API's I can do this, for old API's I need to retain push the salt to the final position (in case someone used type applications).